### PR TITLE
Add poetry

### PR DESCRIPTION
**Pull request by valmik-patel**
I'm creating this request to start a conversation about adding [Poetry](https://python-poetry.org/) to the project for dependency management.

Poetry allows for easy dependency management while ensuring there are no clashes. I've added the dependencies mentioned in `requirements/management.txt` as dev dependencies to poetry. Dependencies in `requirements/cli.txt` and `requirements/webapp.txt` have been added as optional extras dependencies. And finally, the dependencies in `requirements/main.txt` have been added as the main dependencies for the project.

You can test this locally through the following steps
- Install Poetry 
- Run `poetry install --all-extras` from project root to set up a poetry venv and install all dependencies
- Commands can be run in this venv from project root by adding `poetry run` in front of the command

We can do the following things easily through poetry

- **_Run unit tests_** - We can run unit tests using just a single line of code instead of having to use a script like we currently do. 

`poetry run coverage run -m pytest --cov-report xml:cov.xml --cov-report term`

- **_Build dist files_** - We can easily build dist files using `poetry build`
- **_Publish the package_** - Packages can be published to PyPI using `poetry publish`. This will require [configuring poetry with PyPI credentials](https://python-poetry.org/docs/repositories/).

In addition to all this, we can also automate testing when a PR is made using `GitHub Actions`. Using this method, we can block PRs from being merged if the unit tests fail. `GitHub Actions will also allow us to automatically build and publish packages when a PR is merged. Doing all this setup requires us to add PyPI credentials to GitHub secrets at minimum, and we might need to change some GitHub settings, too.

I wanted to start a conversation using this MR so we can get some opinions from the original maintainers. We will need someone with admin access to the repo to help us add secrets and block PRs from merging when the unit tests fail.

@sgpjesus FYI

### CODE REVIEW

**Comment by sgpjesus** [comment]
Hey @jesteria! Can you give your opinion on changing the setup of aequitas to Poetry? Thank you!

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+authors = ["None"]
+readme = "README.md"
+packages = [
+    { include = "aequitas", from="src" },
+    { include = "aequitas_cli", from="src" },
+    { include = "aequitas_webapp", from="src" },
+]
+
+[tool.poetry.dependencies]
+python = "3.9.16"
```
Comment (sgpjesus) [inline pyproject.toml:14]: I've tried installing on my side, and this was giving trouble, I changed it to: `python = ">=3.8,<4"`, some package had a requirement to be less than python version 4 somehow :sweat_smile:

File: pyproject.toml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+[tool.poetry]
+name = "aequitas"
+version = "0.1.0"
```
Comment (sgpjesus) [inline pyproject.toml:3]: `version = "1.0.0"`

File: pyproject.toml [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+
+[tool.poetry.group.dev.dependencies]
+argcmdr = "0.6.0"
+awsebcli = "3.20.10"
+bumpversion = "0.5.3"
+flake8 = "3.5.0"
+twine = "1.14.0"
+pytest = "^8.2.0"
+pytest-cov = "^5.0.0"
+
```
[the remark this answers is not in the document]
Comment (sgpjesus) [reply pyproject.toml:51]:
```suggestion
[[tool.poetry.source]]
name = "pypi-public"
url = "https://pypi.org/simple/"

```

**Review by sgpjesus — CHANGES_REQUESTED** [review]
I believe that with the changes I sent, this MR is good to be merged to master. I think we can also drop the previous requirement files.

**Comment by valmik-patel** [comment]
@sgpjesus I accepted your suggestions and removed the old requirements. Let me know if you need me to do anything else.

**Comment by sgpjesus** [comment]
@valmik-patel Can you just update the `.lock` file, before merging? I think that will be the last thing! I can't add any commits from my side or else my approval will not be valid for the merge.

**Review by sgpjesus — APPROVED** [review]

**Comment by sgpjesus** [comment]
I will actually open a new MR to try and relax some of the dependency versions

**Pull request merged**